pod-operation/src/main.rs

```diff
@@ -2,39 +2,167 @@ use axum::Server;
 use socketioxide::{extract::SocketRef, SocketIo};
 use tracing::{error, info};
 use tracing_subscriber::FmtSubscriber;
-
+mod state;
 mod components;
-mod demo;
+// mod demo;
 mod handlers;
+// use crate::components::pressure_transducer::PressureTransducer;
+// use crate::components::signal_light::SignalLight;
+
+use std::{sync::{Arc, Mutex}, thread, time::Duration};
+use state::State;
+use handlers::{modify_state, read_state};
+use std::collections::HashMap;
+
+struct EnterActions {
+    enter_actions: HashMap<State, fn()>, 
+}
+
+impl EnterActions {
+    fn new() -> Self {
+        let mut enter_actions = HashMap::new();
+        let mut enter_actions_map = HashMap::new();
+        enter_actions_map.insert(State::Init, EnterActions::_enter_init as fn());
+		enter_actions_map.insert(State::Start, EnterActions::_enter_start as fn());
+		enter_actions_map.insert(State::Stop, EnterActions::_enter_stop as fn());
+		enter_actions_map.insert(State::ForceStop, EnterActions::_enter_forcestop as fn());
+		enter_actions_map.insert(State::Load, EnterActions::_enter_load as fn());
+        enter_actions.extend(enter_actions_map);
+		EnterActions {
+            enter_actions,
+        }
+    }
+
+	fn _enter_init() {
+        println!("Entering INIT state");
+    }
+
+    fn _enter_load() {
+        println!("Entering LOAD state");
+		modify_state(State::Init);
+    }
+
+    fn _enter_start() {
+        println!("Entering START state");
+    }
+
+    fn _enter_stop() {
+        println!("Entering STOP state");
+		modify_state(State::Init);
+		println!("State changed to {:?}", read_state());
+    }
+
+	fn _enter_forcestop() {
+        println!("Entering FORCESTOP state");
+		modify_state(State::Init);
+    }
+
+	fn enter_state(&self, state: &State) {
+		println!("Entering state: {:?}", state);
+		if let Some(enter_action) = self.enter_actions.get(state) {
+			enter_action();
+		} else {
+			println!("No enter action defined for {:?}", state);
+		}
+	}
+}
+struct StateTransitions {
+    _state_transitions: HashMap<State, fn()>,
+}
+
+impl StateTransitions {
+	fn new() -> Self {
+		let mut _state_transitions = HashMap::new();
+		_state_transitions.insert(State::Start, Self::_running_periodic as fn());
+		_state_transitions.insert(State::Init, Self::_init_periodic as fn());
+		Self {
+			_state_transitions,
+		}
+	}
+
+	fn _running_periodic() {
+		//println!("Rolling START state");
+	}
+	 
+	fn _init_periodic() {
+		//println!("Rolling INIT state");
+	}
+}
+
 
-use crate::components::pressure_transducer::PressureTransducer;
-use crate::components::signal_light::SignalLight;
 
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
-	tracing::subscriber::set_global_default(FmtSubscriber::default())?;
+	let mut state_now: Option<State> = Some(State::Init);
+	let last_state = Arc::new(Mutex::new(state_now));
```

Comment:
Question: a bit confused here, could you explain why we have both `state_now` and `GLOBAL_STATE`?

Reply:
Will revisit as part of #21.